### PR TITLE
UCT/BASE/SM: Refactor mm to be a transport.

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by yosefe** [comment]
@alinask pls take a look

**Comment by shamisp** [comment]
autobot build

File: src/uct/sm/mm/mm_pd.h [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        static uct_pd_ops_t pd_ops = { \
+            .close        = (void*)ucs_empty_function, \
+            .query        = uct_mm_pd_query, \
+            .mem_alloc    = uct_mm_mem_alloc, \
+            .mem_free     = uct_mm_mem_free, \
+            .mkey_pack    = uct_mm_mkey_pack, \
+        }; \
+        static uct_pd_t pd = { \
+            .ops          = &pd_ops, \
+            .component    = &uct_sysv_pd \
```
Comment (shamisp) [inline src/uct/sm/mm/mm_pd.h]: Looks like a bug … should be _var ?

File: src/uct/tl/context.h [hunk 2]
```diff
@@ -15,22 +17,34 @@
...
-    ucs_status_t           (*rkey_unpack)(const void *rkey_buffer,
+    ucs_status_t           (*rkey_unpack)(uct_pd_component_t *pdc, const void *rkey_buffer,
                                           uct_rkey_t *rkey_p, void **handle_p);
 
-    void                   (*rkey_release)(uct_rkey_t rkey, void *handle);
+    ucs_status_t           (*rkey_release)(uct_pd_component_t *pdc, uct_rkey_t rkey,
+                                           void *handle);
 
     const char             name[UCT_PD_COMPONENT_NAME_MAX];
+    void                   *priv;
```
Comment (shamisp) [inline src/uct/tl/context.h]: Minor - I would name it private or private_data

File: src/uct/sm/mm/mm_pd.c [hunk 3]
```diff
@@ -0,0 +1,100 @@
+/**
+* Copyright (C) Mellanox Technologies Ltd. 2001-2015.  ALL RIGHTS RESERVED.
+*
```
Comment (shamisp) [inline src/uct/sm/mm/mm_pd.c]: UT Battelle copyright, or this is a brand new code ?

File: src/uct/sm/mm/mm_pd.h [hunk 4]
```diff
@@ -0,0 +1,117 @@
+/**
+* Copyright (C) Mellanox Technologies Ltd. 2001-2015.  ALL RIGHTS RESERVED.
```
Comment (shamisp) [inline src/uct/sm/mm/mm_pd.h]: UT-Battelle ?

**Comment by shamisp** [comment]
@yosefe Please see my comments bellow

**Comment by alinask** [comment]
:+1:

**Comment by yosefe** [comment]
@shamisp pls post buildbot failures

**Comment by shamisp** [comment]
@MattBBaker can you please post the error from buildbot. I just don't have an access from my current location.

**Comment by shamisp** [comment]
@yosefe , since @MattBBaker didn't reply, I suggest to go ahead and merge the change. Most likely it is issue on our side (earlier testing passed without problem)

**Pull request merged**